src/symbolicator.index.js

```diff
@@ -0,0 +1,150 @@
+var fs = require('fs');
+
+import { unserializeProfileOfArbitraryFormat } from './profile-logic/process-profile';
+import { SymbolStore } from './profile-logic/symbol-store';
+import {
+    symbolicateProfile,
+    applySymbolicationSteps
+} from './profile-logic/symbolication';
+import * as MozillaSymbolicationAPI from './profile-logic/mozilla-symbolication-api';
+import { SymbolsNotFoundError } from './profile-logic/errors';
+
+var argv = require('minimist')(process.argv.slice(2));
+if (!('input' in argv 
+  && 'output' in argv 
+  && 'server' in argv))
+{
+  console.error("Missing mandatory argument. Usage: symbolicator.js --input <path/to/input.json> --output <path/to/output.json> --server <URI of symbolication server>");
+  process.exit(1);
+}
+
+console.log(`Loading profile from ${argv.input}`);
+var serializedProfile = JSON.parse(fs.readFileSync(argv.input, 'utf8'));
+var profile = await unserializeProfileOfArbitraryFormat(serializedProfile);
+if (profile === undefined) {
+    throw new Error('Unable to parse the profile.');
+}
+
+/**
+ * Simple 'in-memory' symbol DB that conforms to the same interface as SymbolStoreDB but
+ * just stores everything in a simple dictionary instead of IndexedDB. The composite key
+ * [debugName, breakpadId] is flattened to a string "debugName:breakpadId" to use as the
+ * map key.
+ */
+class InMemorySymbolDB {
+
+    _dict: Map<string, SymbolTableAsTuple>
+
+    constructor() {
+        this._dict = new Map();
+    }
+
+    _makeKey(debugName: String, breakpadId: string): string {
+        return `${debugName}:${breakpadId}`;
+    }
+
+    storeSymbolTable(
```

Comment:
nit: use async functions in all these promise-returning methods, for more clarity